### PR TITLE
tweak error message

### LINKED ISSUE #5238: Cross compiling error message about unsupported crates is not helpful
I'm trying to cross-compile Rust applications and libs for bare metal ARM Cortex M3 Blue Pill eval board.
I enter "cargo build --target {some-custom-target-from-.json-specification-file} -vv"
I keep getting error "cannot produce bin for {some-app} as the target {some-custom-target-from-.json-specification-file} does not support these crate types"

With -vv option at least it should be displayed:
1. Where are these crates that are attempted to be built for my app (full path!)
2. What the claimed bad crate type is (what is a "crate type" in the first place?)
3. Possibly links to open issues that explain why this particular crate cannot be built, or what is wrong with the custom target specification

At least with the .json target files we are supposed to have the opportunity to compile for custom archs as long as the LLVM backend supports it.

Kind Regards,

Micel Whave

### PR DESCRIPTION
**Pull request by rleungx**
Closes #5238.

### CODE REVIEW

File: src/cargo/ops/cargo_rustc/context/mod.rs [hunk 1]
```diff
@@ -760,11 +760,11 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
         if ret.is_empty() {
             if !unsupported.is_empty() {
                 bail!(
-                    "cannot produce {} for `{}` as the target `{}` \
-                     does not support these crate types",
-                    unsupported.join(", "),
+                    "cannot compile `{}` package, because target `{}` \
+                     does not support the `{}` crate types",
                     unit.pkg,
-                    self.target_triple()
```
Comment (matklad) [inline src/cargo/ops/cargo_rustc/context/mod.rs]: Let's be extra nice here and do something like

```
bail!("crate type{}", if unsupported.len() == 1 "" else "s")
```
?

**Comment by rleungx** [comment]
@matklad  It seems that `967f094` should be approved. I rebased the commit and sorry for that.

**Pull request merged**